### PR TITLE
adding doxygen/1.8.18

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **doxygen/1.8.18**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
```
/tmp/c3ipr/pr_2071_1_0_0/.conan/data/doxygen/1.8.18/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/bin/doxygen: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /tmp/c3ipr/pr_2071_1_0_0/.conan/data/doxygen/1.8.18/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/bin/doxygen)
/tmp/c3ipr/pr_2071_1_0_0/.conan/data/doxygen/1.8.18/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/bin/doxygen: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /tmp/c3ipr/pr_2071_1_0_0/.conan/data/doxygen/1.8.18/_/_/package/4db1be536558d833e52e862fd84d64d75c2b3656/bin/doxygen)
CMake Warning at /usr/share/cmake-3.16/Modules/FindDoxygen.cmake:447 (message):
  Unable to determine doxygen version: 1
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindDoxygen.cmake:610 (_Doxygen_find_doxygen)
  CMakeLists.txt:7 (find_package)
```

**Comment by prince-chrismc** [comment]
```
ERROR: doxygen/1.8.18: Error in build() method, line 59
	self._unpack_dmg(dest_file)
while calling '_unpack_dmg', line 31
	"Frameworks", "libclang.dylib"), self.build_folder)
	FileNotFoundError: [Errno 2] No such file or directory: '/tmp/c3ipr/pr_2071_4_2_0/.conan/data/doxygen/1.8.18/_/_/build/46f53f156846659bf39ad6675fa0ee8156e859fe/mnt/Doxygen.app/Contents/Frameworks/libclang.dylib'
```

File: recipes/doxygen/all/conanfile.py [hunk 1]
```diff
@@ -11,22 +11,24 @@ class DoxygenInstallerConan(ConanFile):
...
+    settings = "os", "arch", "compiler"
 
-    settings = "os", "arch",
-
-    def config(self):
+    def configure(self):
         if self.settings.os in ["Linux", "Macos"] and self.settings.arch == "x86":
             raise ConanInvalidConfiguration("Doxygen is not supported on {}-{}".format(self.os, self.arch))
+        if tools.Version(self.version) >= "1.8.18" and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "6":
+            raise ConanInvalidConfiguration("Doxygen requires GLIBCXX_3.4.22")
```
Comment (madebr) [inline recipes/doxygen/all/conanfile.py:20]: The error
```
doxygen: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found
```
has nothing to do with the compiler or its compiler version.

What this means is that the downloaded executable is linked against a newer [GNU C library glibc](https://www.gnu.org/software/libc/). Some build bots have an older library installed, which gives rise to the error above.

As of now, conan does not include the glibc version in the package id.

Comment (prince-chrismc) [reply recipes/doxygen/all/conanfile.py:20]: https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html I am not sure what you mean by it "has nothing to do with compiler version" it's a direct relationship but there is a connection...

I release that so it's just a question of picking a bot to package this tool. However I was expecting feedback about this. I am not familiar with any best practice for packaging tools.

Comment (madebr) [reply recipes/doxygen/all/conanfile.py:20]: Oh sorry, so now we have a problem with `GLIBC` and with `GLIBCXX`...

Comment (prince-chrismc) [reply recipes/doxygen/all/conanfile.py:20]: Look like it! Next time I'm at my desk ill google for it

**Comment by KepptnKool** [comment]
Does this part make sense?
`
if tools.Version(self.version) >= "1.8.18" and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "8":
            raise ConanInvalidConfiguration("Doxygen requires GLIBCXX_3.4.22 and GLIBC_2.27") # https://abi-laboratory.pro/index.php?view=changelog&l=glibc&v=2.27
`

On my machine I apparently have glibc 2.27:

**Comment by danimtb** [comment]
Probably that should be a warning instead. @prince-chrismc did you add that invalid configuration due to any limitation in CCI?

**Comment by KepptnKool** [comment]
Doesn't this also contradict the recent [blog post about the build images](https://blog.conan.io/2020/06/17/Conan-Docker-Images.html)?

**Comment by danimtb** [comment]
@KepptnKool the blog post describes the current status and the approach we are taking to solve it in the images, but it says that will be adopted in ConanCenter when possible in the future

**Comment by KepptnKool** [comment]
@danimtb Thanks for clarifying that.

**Comment by prince-chrismc** [comment]
> Probably that should be a warning instead. @prince-chrismc did you add that invalid configuration due to any limitation in CCI?

Yes it is, this recipe downloads a binary so in order to test the image needs a high enough runtime version, since runtimes increase with compiler version, I _tricked_ the CI to use a higher image

In the last PR, it was mentioned we should be building from source... which would solve this problem. However the list of deps is still painful.

**Comment by prince-chrismc** [comment]
GCC 7.5.0 could have the expected runtime version, I picked 8 because I was certain it would always work. I dont think GCC 7.4 and less could have the required version. I was hesitant between 7 and 8. If people install with 7 and is doesnt work, it would also be an issue.

The [supported platforms](https://github.com/conan-io/conan-center-index/wiki/Supported-Platforms-And-Configurations#linux) does not list the minor for GCC compilers, @KepptnKool feel free to open a PR and try to lower the GCC requirement to 7.5 see if it pass with CCI

**Pull request merged**